packages/crayons-core/src/components/data-table/data-table.tsx

```diff
@@ -390,6 +369,318 @@ export class DataTable {
     return this.isLoading;
   }
 
+  /**
+   * Private
+   * getColumnsState function to call when updating a column state.
+   * Updating state using this function can help update the UI easily.
+   * @param state current state to modify
+   * @param stateKey key that can help identify what object in state to identify
+   * @param keyValuePairs key value pair to modify the object
+   * @returns modifies state object
+   */
+  getColumnsState(state, stateKey, keyValuePairs) {
+    let currentElementIndex;
+    const currentElement = state.filter((stateElement, elementIndex) => {
+      const isCurrentElement = stateElement.key === stateKey;
+      if (isCurrentElement) {
+        currentElementIndex = elementIndex;
+      }
+      return isCurrentElement;
+    })[0];
+    const modifiedCurrentElement = { ...currentElement, ...keyValuePairs };
+    const otherElements = state.filter(
+      (stateElement) => stateElement.key !== stateKey
+    );
+    const preElements = otherElements.slice(0, currentElementIndex);
+    const postElements = otherElements.slice(
+      currentElementIndex,
+      otherElements.length
+    );
+    if (modifiedCurrentElement.position !== currentElement.position) {
+      preElements.map((preElement) => {
```

Comment:
it's better to avoid mutating the `preElement` and `postElement`.

Reply:
preElement, postElement are slices of original param, hence not mutating the original param. Also, changed map to for loop to avoid any confusion.